### PR TITLE
Albrja/mic 5383/psimulate clock configuration

### PR DESCRIPTION
**Pull request by albrja**
### Updates how vivarium work horse gets number of simulation steps
- *Category*: Feature
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5383

### Changes and notes
-Updates vivarium work horse method to get number of steps from InteractiveContext method

### Testing
Successfully ran parallel simulation with custom clock

### CODE REVIEW

File: src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py [hunk 1]
```diff
@@ -93,9 +93,7 @@ def work_horse(job_parameters: dict) -> Tuple[pd.DataFrame, Dict[str, pd.DataFra
                 f'Simulant initialization completed in {exec_time["simulant_initialization_minutes"]:.3f} minutes.'
             )
             start_time = pd.Timestamp(**sim.configuration.time.start.to_dict())
-        end_time = pd.Timestamp(**sim.configuration.time.end.to_dict())
-        step_size = pd.Timedelta(days=sim.configuration.time.step_size)
-        num_steps = int(math.ceil((end_time - start_time) / step_size))
+        num_steps = sim.get_number_of_steps(start_time)
```
Comment (rmudambi) [inline src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py:96]: We shouldn't be providing start time to this method.

**Review by rmudambi — CHANGES_REQUESTED** [review]

File: src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py (hunk 1 above)
Comment (albrja) [reply src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py:96]: But we have to get it from the Parallel sim context or the backups

File: CHANGELOG.rst [hunk 2]
```diff
@@ -1,3 +1,7 @@
+**2.0.4 - TBD**
```
Comment (stevebachmeier) [inline CHANGELOG.rst:1]: just a reminder to change the date here

File: src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py (hunk 1 above)
Comment (stevebachmeier) [reply src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py:96]: On the line above, it gets the start time from `sim.configuration.time.start` Isn't that just the model spec start time? Or does this value actually change in the event you're starting from a backup?

**Review by stevebachmeier — CHANGES_REQUESTED** [review]

**Review by rmudambi — APPROVED** [review]

**Review by stevebachmeier — APPROVED** [review]

**Review by patricktnast — APPROVED** [review]

**Pull request merged**